packages/block-editor/src/components/block-card/index.js

```diff
@@ -166,20 +193,43 @@ function BlockCard( {
 						: undefined
 				}
 			>
-				<BlockIcon icon={ icon } showColors />
-				<VStack spacing={ 1 }>
-					<TitleElement className="block-editor-block-card__title">
-						<span className="block-editor-block-card__name">
-							{ !! name?.length ? name : title }
-						</span>
-						{ ! parentClientId && ! isChild && !! name?.length && (
-							<Badge>{ title }</Badge>
-						) }
-					</TitleElement>
-					{ ! parentClientId && ! isChild && description && (
-						<Text className="block-editor-block-card__description">
-							{ description }
-						</Text>
+				<VStack spacing={ 4 }>
+					<HStack spacing={ 1 } align="start">
+						<BlockIcon icon={ icon } showColors />
+						<VStack spacing={ 1 }>
+							<TitleElement className="block-editor-block-card__title">
+								<span className="block-editor-block-card__name">
+									{ !! name?.length ? name : title }
+								</span>
+								{ ! parentClientId &&
+									! isChild &&
+									!! name?.length && (
+										<Badge>{ title }</Badge>
+									) }
+							</TitleElement>
+							{ ! parentClientId && ! isChild && description && (
+								<Text className="block-editor-block-card__description">
+									{ description }
+								</Text>
+							) }
+							{ children }
+						</VStack>
+					</HStack>
+					{ ( isBlockHidden || hasHiddenParent ) && (
+						<Notice
+							className="block-editor-block-card__hidden-notice"
+							status="warning"
+							isDismissible={ false }
+						>
+							<HStack spacing={ 2 } justify="start">
+								<Icon icon={ unseen } />
+								<Text>
+									{ hasHiddenParent
```

Comment:
What should happen if both the current block is hidden and the parent is hidden? Currently the parent text overrides the child text.

The example is if you have a hidden block within a parent block that is also hidden.

Reply:
Good question. TBH I hadn't thought of that. I'll look into it. Maybe it should just default to "Block hidden" in that case.

Reply:
I think it's something that's less likely to happen right now, but more likely when we get to the hiding-by-breakpoints behaviour. I.e. I could imagine someone might have a wrapper for "mobile and tablet" (hidden on desktop) and within it, they might have something hidden on mobile, or something hidden on tablet.

Reply:
I think this POC kinda tackled this scenario with variable messages (if I'm understanding you right)

https://github.com/WordPress/gutenberg/pull/73888/files#diff-068538657061c47f0a58cfc331259bbeabe11bcc0c576a0cc496d262e5b0038cR361